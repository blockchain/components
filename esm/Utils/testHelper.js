import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';
import theme from '../Tools/theme';

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(React.createElement(ThemeProvider, {
    theme: theme
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

export function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [shallow].concat(Array.prototype.slice.call(arguments)));
}
export function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [mount].concat(Array.prototype.slice.call(arguments)));
}