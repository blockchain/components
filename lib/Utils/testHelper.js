"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.shallowWithTheme = shallowWithTheme;
exports.mountWithTheme = mountWithTheme;

var _react = _interopRequireDefault(require("react"));

var _enzyme = require("enzyme");

var _styledComponents = require("styled-components");

var _theme = _interopRequireDefault(require("../Tools/theme"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function wrapWithTheme(fn, children, options) {
  var wrapper = fn(_react.default.createElement(_styledComponents.ThemeProvider, {
    theme: _theme.default
  }, children), options);
  return wrapper[fn.name]({
    context: wrapper.instance().getChildContext()
  });
}

function shallowWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.shallow].concat(Array.prototype.slice.call(arguments)));
}

function mountWithTheme() {
  return wrapWithTheme.apply(void 0, [_enzyme.mount].concat(Array.prototype.slice.call(arguments)));
}