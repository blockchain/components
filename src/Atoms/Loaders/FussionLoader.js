// @flow strict
import React from 'react'
import styled, { keyframes } from 'styled-components'

const Wrapper = styled.div`
  height: 40px;
  width: 40px;
  display: flex;
  align-items: center;
  justify-content: center;
`

const ball1 = keyframes`
  0% {
    opacity: 0.35;
  }

  50% {
    top: -100%;
    left: 200%;
    opacity: 1;
  }

  100% {
    top: 50%;
    left: 100%;
    z-index: 2;
    opacity: 0.35;
  }
`

const ball2 = keyframes`
  0% {
    opacity: 0.35;
  }

  50% {
    top: 200%;
    left: 200%;
    opacity: 1;
  }

  100% {
    top: 100%;
    left: 50%;
    z-index: 1;
    opacity: 0.35;
  }
`

const ball3 = keyframes`
  0% {
    opacity: 0.35;
  }

  50% {
    top: 200%;
    left: -100%;
    opacity: 1;
  }

  100% {
    top: 50%;
    left: 0;
    z-index: 2;
    opacity: 0.35;
  }
`

const ball4 = keyframes`
  0% {
    opacity: 0.35;
  }

  50% {
    top: -100%;
    left: -100%;
    opacity: 1;
  }

  100% {
    top: 0;
    left: 50%;
    z-index: 1;
    opacity: 0.35;
  }
`

const Animation = styled.div`
  width: 8px;
  height: 8px;
  position: relative;
  box-sizing: border-box;

  & > div {
    position: absolute;
    display: inline-block;
    box-sizing: border-box;
    width: 16px;
    height: 16px;
    border-radius: 100%;
    background-color: ${(props) => props.theme['loaderColor']};
    transform: translate(-50%, -50%);
  }

  & > div:nth-child(1) {
    top: 0;
    left: 50%;
    z-index: 1;
    animation: ${ball1} 1s 0s ease infinite;
  }

  & > div:nth-child(2) {
    top: 50%;
    left: 100%;
    z-index: 2;
    animation: ${ball2} 1s 0s ease infinite;
  }

  & > div:nth-child(3) {
    top: 100%;
    left: 50%;
    z-index: 1;
    animation: ${ball3} 1s 0s ease infinite;
  }

  & > div:nth-child(4) {
    top: 50%;
    left: 0;
    z-index: 2;
    animation: ${ball4} 1s 0s ease infinite;
  }
`

const FussionLoader = () => (
  <Wrapper>
    <Animation>
      <div />
      <div />
      <div />
      <div />
    </Animation>
  </Wrapper>
)

export default FussionLoader
