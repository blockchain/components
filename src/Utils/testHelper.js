import { mount } from 'enzyme'
import { ThemeConsumer } from 'styled-components'
import theme from '../Tools/theme'

export const mountWithTheme = (children) => {
  ThemeConsumer._currentValue = theme
  return mount(children)
}
