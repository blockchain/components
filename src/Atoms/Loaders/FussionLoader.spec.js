import React from 'react'
import { mountWithTheme } from '../../Utils/testHelper'

import FussionLoader from './FussionLoader'

describe('FussionLoader', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<FussionLoader />)

    expect(tree.find(FussionLoader)).toMatchSnapshot()
  })
})
