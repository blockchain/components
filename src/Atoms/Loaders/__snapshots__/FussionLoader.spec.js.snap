// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FussionLoader should render correctly 1`] = `
.c0 {
  height: 40px;
  width: 40px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 8px;
  height: 8px;
  position: relative;
  box-sizing: border-box;
}

.c1 > div {
  position: absolute;
  display: inline-block;
  box-sizing: border-box;
  width: 16px;
  height: 16px;
  border-radius: 100%;
  background-color: #10ADE4;
  -webkit-transform: translate(-50%,-50%);
  -ms-transform: translate(-50%,-50%);
  transform: translate(-50%,-50%);
}

.c1 > div:nth-child(1) {
  top: 0;
  left: 50%;
  z-index: 1;
  -webkit-animation: cDEyoB 1s 0s ease infinite;
  animation: cDEyoB 1s 0s ease infinite;
}

.c1 > div:nth-child(2) {
  top: 50%;
  left: 100%;
  z-index: 2;
  -webkit-animation: bjqCJF 1s 0s ease infinite;
  animation: bjqCJF 1s 0s ease infinite;
}

.c1 > div:nth-child(3) {
  top: 100%;
  left: 50%;
  z-index: 1;
  -webkit-animation: IydCq 1s 0s ease infinite;
  animation: IydCq 1s 0s ease infinite;
}

.c1 > div:nth-child(4) {
  top: 50%;
  left: 0;
  z-index: 2;
  -webkit-animation: dsyxXP 1s 0s ease infinite;
  animation: dsyxXP 1s 0s ease infinite;
}

<FussionLoader>
  <styled.div>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  height: 40px;
  width: 40px;
  display: flex;
  align-items: center;
  justify-content: center;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="c0"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-bwzfXH",
                  "isStatic": false,
                  "lastClassName": "c1",
                  "rules": Array [
                    "
  width: 8px;
  height: 8px;
  position: relative;
  box-sizing: border-box;

  & > div {
    position: absolute;
    display: inline-block;
    box-sizing: border-box;
    width: 16px;
    height: 16px;
    border-radius: 100%;
    background-color: ",
                    [Function],
                    ";
    transform: translate(-50%, -50%);
  }

  & > div:nth-child(1) {
    top: 0;
    left: 50%;
    z-index: 1;
    animation: ",
                    Keyframes {
                      "id": "sc-keyframes-cDEyoB",
                      "inject": [Function],
                      "name": "cDEyoB",
                      "rules": Array [
                        "@-webkit-keyframes cDEyoB{0%{opacity:0.35;}50%{top:-100%;left:200%;opacity:1;}100%{top:50%;left:100%;z-index:2;opacity:0.35;}}",
                        "@keyframes cDEyoB{0%{opacity:0.35;}50%{top:-100%;left:200%;opacity:1;}100%{top:50%;left:100%;z-index:2;opacity:0.35;}}",
                      ],
                      "toString": [Function],
                    },
                    " 1s 0s ease infinite;
  }

  & > div:nth-child(2) {
    top: 50%;
    left: 100%;
    z-index: 2;
    animation: ",
                    Keyframes {
                      "id": "sc-keyframes-bjqCJF",
                      "inject": [Function],
                      "name": "bjqCJF",
                      "rules": Array [
                        "@-webkit-keyframes bjqCJF{0%{opacity:0.35;}50%{top:200%;left:200%;opacity:1;}100%{top:100%;left:50%;z-index:1;opacity:0.35;}}",
                        "@keyframes bjqCJF{0%{opacity:0.35;}50%{top:200%;left:200%;opacity:1;}100%{top:100%;left:50%;z-index:1;opacity:0.35;}}",
                      ],
                      "toString": [Function],
                    },
                    " 1s 0s ease infinite;
  }

  & > div:nth-child(3) {
    top: 100%;
    left: 50%;
    z-index: 1;
    animation: ",
                    Keyframes {
                      "id": "sc-keyframes-IydCq",
                      "inject": [Function],
                      "name": "IydCq",
                      "rules": Array [
                        "@-webkit-keyframes IydCq{0%{opacity:0.35;}50%{top:200%;left:-100%;opacity:1;}100%{top:50%;left:0;z-index:2;opacity:0.35;}}",
                        "@keyframes IydCq{0%{opacity:0.35;}50%{top:200%;left:-100%;opacity:1;}100%{top:50%;left:0;z-index:2;opacity:0.35;}}",
                      ],
                      "toString": [Function],
                    },
                    " 1s 0s ease infinite;
  }

  & > div:nth-child(4) {
    top: 50%;
    left: 0;
    z-index: 2;
    animation: ",
                    Keyframes {
                      "id": "sc-keyframes-dsyxXP",
                      "inject": [Function],
                      "name": "dsyxXP",
                      "rules": Array [
                        "@-webkit-keyframes dsyxXP{0%{opacity:0.35;}50%{top:-100%;left:-100%;opacity:1;}100%{top:0;left:50%;z-index:1;opacity:0.35;}}",
                        "@keyframes dsyxXP{0%{opacity:0.35;}50%{top:-100%;left:-100%;opacity:1;}100%{top:0;left:50%;z-index:1;opacity:0.35;}}",
                      ],
                      "toString": [Function],
                    },
                    " 1s 0s ease infinite;
  }
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-bwzfXH",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="c1"
            >
              <div />
              <div />
              <div />
              <div />
            </div>
          </StyledComponent>
        </styled.div>
      </div>
    </StyledComponent>
  </styled.div>
</FussionLoader>
`;
